scripts/scripts.js

```diff
@@ -621,7 +621,13 @@ window.store = new (class {
  * Generates HTML for the premium article banner.
  * @param {Number} Number of leftover articles to compare to.
  */
-export const premiumArticleBanner = (leftoverArticles = 0) => {
+export const premiumArticleBanner = (customLeftoverArticles = null) => {
+  let leftoverArticles = customLeftoverArticles;
+  if (typeof customLeftoverArticles !== 'number') {
+    leftoverArticles = Number(window.sessionStorage.freeArticles || 3);
+  }
+  window.sessionStorage.freeArticles = Math.max(leftoverArticles - 1, 0);
```

Comment:
sessionStorage can be disabled by the user (by default disabled in Chrome incognito mode AFAIR) so maybe we should rely on something else eg use the `window.name` trick WDYT ?

Reply:
I changed to use window.name